src/ledger/InternalLedgerEntry.h

```diff
@@ -46,7 +46,16 @@ class InternalLedgerKey
 
     void checkDiscriminant(InternalLedgerEntryType expected) const;
 
+    void type(InternalLedgerEntryType t);
+    LedgerKey& ledgerKeyRef();
```

Comment:
These are only ever used for assignment. Should we change them to setters, for example
```c++
void setLedgerKey(LedgerKey const& key);
void setLedgerKey(LedgerKey&& key);
```
This would make it even harder for a future maintainer to accidentally invalidate the hash by using a reference improperly.

Reply:
That's fine for now. If we ever change the way those methods are used (unlikely) we can change to a safer idiom.